gen.go

```diff
@@ -166,10 +192,84 @@ func typeName(pkg string, t reflect.Type) string {
 			// It's a built-in.
 			return t.String()
 		} else if pkgPath == pkg {
-			return t.Name()
+			return genericName(t.Name())
+		}
+		return fmt.Sprintf("%s.%s", resolvePkgName(pkgPath, t.String()), genericName(t.Name()))
+	}
+}
+
+func getGenericTypeParameter(index int) string {
+	parameterLetters := "TUVWXYZ"
+	if index < len(parameterLetters) {
+		return string(parameterLetters[index])
+	}
+	return "T" + strconv.Itoa(index-len(parameterLetters)+1)
+}
```

Comment:
Honestly, maybe just always use `T1..TN`?

Reply:
I want this to conform to how the user is defining their type, and they'll most likely be sticking to convention of `T, U, V ...` and probably not getting further than 1-2 types. If we go `T1` then I think we enforce that on users because I think they need to be consistent across a type.

Reply:
I thought these were internal to the generated code. In general, the "conventions" for placeholders tend to be pretty loose.

Reply:
I thought that it needed to be consistent across methods of the same type, but I just tested it out elsewhere and apparently it can be whatever you like as long as it's consistent with in a single method. So `Tn` is fine.